File: Jumpa 2/japa1/RekeningBank.java
public class RekeningBank {
    private String nomorRekening;
    private String namaPemilik;
    private double saldo;

    public RekeningBank(String nomorRekening, String namaPemilik, double saldo) {
        if (nomorRekening == null || nomorRekening.isEmpty()) {
            throw new IllegalArgumentException("Nomor rekening tidak boleh kosong.");
        }

        if (saldo < 0) {
            throw new IllegalArgumentException("Saldo tidak boleh kurang dari 0.");
        }

        this.nomorRekening = nomorRekening;
        this.namaPemilik = namaPemilik;
        this.saldo = saldo;
    }

    public void tampilkanInfo() {
        System.out.println("Nomor Rekening : " + nomorRekening);
        System.out.println("Nama Pemilik   : " + namaPemilik);
        System.out.println("Saldo          : Rp" + saldo);
    }

    public void setor(double jumlah) {
        if (jumlah <= 0) {
            throw new IllegalArgumentException(
                "Jumlah setor harus lebih dari 0."
            );
        }

        saldo += jumlah;
    }

    public void tarik(double jumlah) {
        if (jumlah <= 0) {
            throw new IllegalArgumentException(
                "Jumlah penarikan harus lebih dari 0."
            );
        }

        if (jumlah > saldo) {
            throw new IllegalArgumentException(
                "Saldo tidak mencukupi."
            );
        }

        saldo -= jumlah;
    }
}

File: Jumpa 2/pAp1/RekeningBank.php
<?php

class RekeningBank
{
    private string $nomorRekening;
    private string $namaPemilik;
    private float $saldo;

    public function __construct(
        string $nomorRekening,
        string $namaPemilik,
        float $saldo
    ) {
        if (empty($nomorRekening)) {
            throw new InvalidArgumentException(
                "Nomor rekening tidak boleh kosong."
            );
        }

        if ($saldo < 0) {
            throw new InvalidArgumentException(
                "Saldo tidak boleh kurang dari 0."
            );
        }

        $this->nomorRekening = $nomorRekening;
        $this->namaPemilik = $namaPemilik;
        $this->saldo = $saldo;
    }

    public function tampilkanInfo(): void
    {
        echo "Nomor Rekening : " . $this->nomorRekening . PHP_EOL;
        echo "Nama Pemilik   : " . $this->namaPemilik . PHP_EOL;
        echo "Saldo          : Rp" . $this->saldo . PHP_EOL;
    }

    public function setor(float $jumlah): void
    {
        if ($jumlah <= 0) {
            throw new InvalidArgumentException(
                "Jumlah setor harus lebih dari 0."
            );
        }

        $this->saldo += $jumlah;
    }

    public function tarik(float $jumlah): void
    {
        if ($jumlah <= 0) {
            throw new InvalidArgumentException(
                "Jumlah penarikan harus lebih dari 0."
            );
        }

        if ($jumlah > $this->saldo) {
            throw new InvalidArgumentException(
                "Saldo tidak mencukupi."
            );
        }

        $this->saldo -= $jumlah;
    }
}